### PR TITLE
Introduces `ForkChoiceUpdatedV3` for Deneb

### PR DESCRIPTION
**Pull request by tbenr**
- Adds `PayloadAttributesV3` with parent beacon block root
- Adds `ForkChoiceUpdatedV3` and plugs it in the Deneb flow
- Updates `forkChoiceNotifier` to provide the parent block root to our internal `PayloadBuildingAttributes`
- Updates tests

related to #7263

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/executionlayer/src/test/java/tech/pegasys/teku/ethereum/executionlayer/DenebExecutionClientHandlerTest.java [hunk 1]
```diff
@@ -106,21 +106,22 @@ void engineForkChoiceUpdated_shouldCallEngineForkChoiceUpdatedV2() {
             dataStructureUtil.randomEth1Address(),
             Optional.empty(),
             Optional.of(List.of()),
+            dataStructureUtil.randomBytes32(),
```
Comment (StefanBratanov) [inline ethereum/executionlayer/src/test/java/tech/pegasys/teku/ethereum/executionlayer/DenebExecutionClientHandlerTest.java:109]: Just the test name needs to be changed as well for consistency

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/executionlayer/PayloadBuildingAttributes.java [hunk 2]
```diff
@@ -91,12 +97,19 @@ public boolean equals(final Object o) {
...
         && Objects.equals(validatorRegistration, that.validatorRegistration)
-        && Objects.equals(maybeWithdrawals, that.maybeWithdrawals);
+        && Objects.equals(maybeWithdrawals, that.maybeWithdrawals)
+        && Objects.equals(parentBeaconBlockRoot, that.parentBeaconBlockRoot);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(timestamp, prevRandao, feeRecipient, validatorRegistration);
+    return Objects.hash(
```
Comment (StefanBratanov) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/executionlayer/PayloadBuildingAttributes.java:107]: This was missed before I suppose can add blockSlot to equals,hashCode and toString

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ProposersDataManager.java [hunk 3]
```diff
@@ -199,11 +199,17 @@ public SafeFuture<Optional<PayloadBuildingAttributes>> calculatePayloadBuildingA
     return getStateInEpoch(epoch)
         .thenApplyAsync(
             maybeState ->
-                calculatePayloadBuildingAttributes(blockSlot, epoch, maybeState, mandatory),
+                calculatePayloadBuildingAttributes(
+                    forkChoiceUpdateData.getForkChoiceState().getHeadBlockRoot(),
```
Comment (StefanBratanov) [inline ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ProposersDataManager.java]: I would just initialise it before Bytes32 parentBeaconBlockRoot = ... to bring more clarity.

**Review by StefanBratanov — APPROVED** [review]
LGTM with couple of comments

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/executionlayer/PayloadBuildingAttributes.java (hunk 2 above)
Comment (tbenr) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/executionlayer/PayloadBuildingAttributes.java:107]: ye noticed that too.. for some reason I ended up not adding it

**Pull request merged**